cmd/kubeapps-apis/plugins/resources/v1alpha1/server.go

```diff
@@ -49,9 +54,45 @@ type Server struct {
 	// corePackagesClientGetter holds a function to obtain the core.packages.v1alpha1
 	// client. It is similarly initialised in NewServer() below.
 	corePackagesClientGetter func() (pkgsGRPCv1alpha1.PackagesServiceClient, error)
+
+	// We keep a restmapper to cache discovery of REST mappings from GVK->GVR.
+	restMapper meta.RESTMapper
+
+	// kindToResource is a function to convert a GVK to GVR. Can be replaced
+	// in tests with a dummy version using the unsafe helpers while the real
+	// implementation queries the k8s API for a REST mapper.
+	kindToResource func(meta.RESTMapper, schema.GroupVersionKind) (schema.GroupVersionResource, error)
+}
+
+// createRESTMapper returns a rest mapper configured with the APIs of the
+// local k8s API server. This is used to convert between the GroupVersionKinds
+// of the resource references to the GroupVersionResource used by the API server.
+func createRESTMapper() (meta.RESTMapper, error) {
```

Comment:
This is nuts... what an extra amount of coding for performing just a conversion...I guess I lack some K8s internal knowledge here, but I don't know why we need this kind of runtime information.
Thanks a lot for coming up with this solution (I don't know if the restmapper might be useful for additional things as well).

Reply:
It comes down to the fact that a client (`kubectl` or in our case our `client-go` client) cannot safely make assumptions about the mapping between an object (the yaml manifest with the GroupVersionKind) and the resource endpoint in the API. The only safe way to determine the mapping is to query the api server for the API information, which is what this now does.